entities/database/models/file.py

```diff
@@ -43,11 +67,33 @@ class File(Base):
 @event.listens_for(File, "before_delete")
 def before_delete(mapper: Mapper, connection: Connection, target: File) -> None:
     """
-    Before deleting a File object, make sure to scrub the foreign keys in the Entity it's associated with.
+    Before deleting a File object, check whether we need to delete it from S3, and
+    make sure to scrub the foreign keys in the Entity it's associated with.
     """
+    table_files = target.__table__
     table_entity = target.entity.__table__
-    values = {f"{target.entity_field_name}_id": None}
+    settings = File.get_settings()
+    s3_client = File.get_s3_client()
```

Comment:
This is because this event is not triggered within a FastAPI context we can fetch from